### PR TITLE
Use varchar(limit) in Redis and Kafka connectors

### PR DESCRIPTION
**Pull request by losipiuk**
@arhimondr please give it a first read

### CODE REVIEW

**Comment by losipiuk** [comment]
@arhimondr ping. Please take a look before I pass it to FB.

File: presto-redis/src/test/java/com/facebook/presto/redis/util/RedisLoader.java [hunk 1]
```diff
@@ -149,7 +149,7 @@ private Object convertValue(Object value, Type type)
                 return null;
             }
 
-            if (BOOLEAN.equals(type) || VARCHAR.equals(type)) {
+            if (BOOLEAN.equals(type) || type instanceof VarcharType) {
```
Comment (arhimondr) [inline presto-redis/src/test/java/com/facebook/presto/redis/util/RedisLoader.java]: you can use `Varchars.isVarcharType(Type)`

File: presto-record-decoder/src/main/java/com/facebook/presto/decoder/json/JsonFieldDecoder.java [hunk 2]
```diff
@@ -122,7 +124,14 @@ public double getDouble()
         public Slice getSlice()
         {
             String textValue = value.isValueNode() ? value.asText() : value.toString();
-            return isNull() ? EMPTY_SLICE : utf8Slice(textValue);
+            if (isNull()) {
+                return EMPTY_SLICE;
+            }
+            Slice slice = utf8Slice(textValue);
+            if (columnHandle.getType() instanceof VarcharType) {
```
Comment (arhimondr) [inline presto-record-decoder/src/main/java/com/facebook/presto/decoder/json/JsonFieldDecoder.java]: `Varchars.isVarcharType()`

File: presto-record-decoder/src/main/java/com/facebook/presto/decoder/csv/CsvFieldDecoder.java [hunk 3]
```diff
@@ -92,7 +94,14 @@ public double getDouble()
             @Override
             public Slice getSlice()
             {
-                return isNull() ? EMPTY_SLICE : utf8Slice(value);
+                if (isNull()) {
+                    return EMPTY_SLICE;
+                }
+                Slice slice = utf8Slice(value);
+                if (columnHandle.getType() instanceof VarcharType) {
```
Comment (arhimondr) [inline presto-record-decoder/src/main/java/com/facebook/presto/decoder/csv/CsvFieldDecoder.java]: `isVarcharType()`

File: presto-record-decoder/src/main/java/com/facebook/presto/decoder/raw/RawFieldDecoder.java [hunk 4]
```diff
@@ -220,7 +222,11 @@ public Slice getSlice()
             }
 
             if (fieldType == FieldType.BYTE) {
-                return Slices.wrappedBuffer(value.slice());
+                Slice slice = Slices.wrappedBuffer(value.slice());
+                if (columnHandle.getType() instanceof VarcharType) {
```
Comment (arhimondr) [inline presto-record-decoder/src/main/java/com/facebook/presto/decoder/raw/RawFieldDecoder.java]: `isVarcharType`

File: presto-kafka/src/test/java/com/facebook/presto/kafka/util/KafkaLoader.java [hunk 5]
```diff
@@ -126,7 +126,7 @@ private Object convertValue(Object value, Type type)
                 return null;
             }
 
-            if (BOOLEAN.equals(type) || VARCHAR.equals(type)) {
+            if (BOOLEAN.equals(type) || type instanceof VarcharType) {
```
Comment (arhimondr) [inline presto-kafka/src/test/java/com/facebook/presto/kafka/util/KafkaLoader.java]: `isVarcharType`

**Comment by arhimondr** [comment]
LGTM

**Comment by losipiuk** [comment]
superseded by #5105

**Pull request closed without merging**